xla/service/gpu/ir_emitter_triton_rocm.cc

```diff
@@ -69,47 +72,50 @@ absl::Status CreateTritonPipeline(
   pm.addPass(mlir::createSymbolDCEPass());
 
   // Based on make_ttgir() in
-  // @triton//:third_party/nvidia/backend/compiler.py
+  // @triton//:third_party/amd/backend/compiler.py
   pm.addPass(mt::createConvertTritonToTritonGPUPass(
-      absl::StrFormat("cuda:%u", ccAsInt), block_level_parameters.num_warps,
+      absl::StrCat("hip:", ccRocm.gfx_version()),
+      block_level_parameters.num_warps,
       threadsPerWarp, block_level_parameters.num_ctas));
   pm.addPass(mt::gpu::createTritonGPUCoalesce());
   pm.addPass(mt::gpu::createTritonGPURemoveLayoutConversions());
   pm.addPass(mt::gpu::createTritonGPUOptimizeThreadLocality());
-  pm.addPass(createSparseBlockedToMMAPass());
   pm.addPass(mt::gpu::createTritonGPUAccelerateMatmul());
   pm.addPass(mt::gpu::createTritonGPURemoveLayoutConversions());
   // TODO ROCm Check if we want to compare MI100 and greater
+  pm.addPass(mlir::createTritonAMDGPUOptimizeEpiloguePass());
   pm.addPass(mt::gpu::createTritonGPUOptimizeDotOperands({true}));
-  pm.addPass(mlir::createCSEPass());
-  pm.addPass(
-      mt::gpu::createTritonGPUPipeline({block_level_parameters.num_stages}));
-  pm.addPass(mt::gpu::createTritonGPUPrefetch());
-
-  // TODO ROCm Check if we want to compare MI100 and greater
+  if(block_level_parameters.num_stages == 0 &&
```

Comment:
What's the semantic meaning of `0` stages here and below?

This doesn't seem like it should be a valid value.

Reply:
Was using the same check as in corresponding Triton code:
https://github.com/openxla/triton/blob/7c6c2ef26430d34bfec657d6e2d957505425e577/third_party/amd/backend/compiler.py#L151

I can investigate this further to get understanding of semantic meaning.

Reply:
Value zero represents AMD specific software pipelining / double buffering.

Reply:
Thanks for clarifying that this is indeed intentional. I think that if this is in our codebase, we need to understand exactly what the implications are so that people unfamiliar with the code can maintain it. I think this is not enough as is, because the semantics are still not clear.

I would like us to do the following:

1. let's give a name to `0`, i.e. make it a `constexpr int <some semantically meaningful name> = 0;`;
2. comment on top of the definition what exactly `0` means here. Presumably, this triggers some specific behaviour that is not expressible using other values for `num_stages`---what is it?